src/command_modules/azure-cli-iot/azure/cli/command_modules/iot/_params.py

```diff
@@ -16,16 +16,29 @@ def get_device_id_completion_list(prefix, action, parsed_args, **kwargs):#pylint
     client = iot_hub_service_factory(kwargs)
     return [d.device_id for d in iot_device_list(client, parsed_args.hub_name, top=100)] if parsed_args.hub_name else []
 
-hub_name_type = CliArgumentType(completer=get_resource_name_completion_list('Microsoft.Devices/IotHubs'),
-                                help='IoT Hub name.')
 
-register_cli_argument('iot hub', 'hub_name', hub_name_type, options_list=('--name', '-n'))
```

Comment:
In general, this approach was the correct approach. For the `iot hub` commands, hub_name should be aliased to --name/-n. For other cases (including any subgroups) it would be 'hub-name'.